### ISSUE TITLE
"--attempt-instant-ddl" flag causing some issues with hooks and is incompatible with --execute flag

### ISSUE THREAD

**Issue by YazanNoon** [github/gh-ost#1365]
when I tried to use the --attempt-instant-ddl flag, I faced two issues:
- the on-success hook will not trigger after they migration is done.
- even if I don't use the --execute flag, the migration will still happen anyway.

**Comment by hakusaro** [comment]
If it's not too much effort @timvaillancourt, I would suggest adding this to the docs around `--attempt-instant-ddl`, as looking at this, the original PR, and other issues led me to pulling this flag out of our production gh-ost scripts for safety.

I know you're very busy, and this is a low priority task, however.

**Comment by luddd3** [comment]
I just also got bitten by the fact that the migration is executed without the "--execute"  flag when "--attempt-instant-ddl" is used.

**Comment by timvaillancourt** [comment]
@YazanNoon thanks for the report. The 2nd issue sounds dangerous so I have made [a fix for that first here](https://github.com/github/gh-ost/pull/1440). If you have time to help validate the fix before/after merge, that would be very helpful 🙇 

cc @hakusaro / @luddd3

**Comment by YazanNoon** [comment]
thanks a lot @timvaillancourt!
the PR looks good!

**Comment by hakusaro** [comment]
Thanks so much! It looks good to me! ^_^

**Comment by luddd3** [comment]
Thank you!

**Comment by meiji163** [comment]
Closing as fixed after #1463
